mp_api/client/client.py

```diff
@@ -177,6 +177,18 @@ def __exit__(self, exc_type, exc_val, exc_tb):
         """
         self.session.close()
 
+    def __getattr__(self, attr):
+        if attr == "alloys":
+            raise MPRestError("Alloy addon package not installed. "
+                              "To query alloy data install the pymatgen-analysis-alloys package.")
+        elif attr == "charge_density":
+            raise MPRestError("boto3 not installed. "
+                              "To query charge density data install the boto3 package.")
+        else:
+            raise AttributeError(
+                    f"{self.__class__.__name__!r} object has no attribute {attr!r}"
+                )
+
```

Comment:
Include URLs to PyPI for optional dependencies?

Reply:
Instead of the URL, I have just added the pip command directly.